### PR TITLE
[MetaSchedule] Enable Adapative Training For XGBoost Cost Model

### PR DESCRIPTION
**Pull request by zxybazh**
CostModel retraining is a time consuming part for MetaSchedule tuning, similar to AutoScheduler, we can alleviate it with an adapative way of increasing waiting period between each retraining. This PR introduced an argument called `adpative_training` in `TuneConfig` and the constructor of `XGBoostModel` to enable the capability. Testing tuning scripts are also updated.

### CODE REVIEW

**Comment by junrushao1994** [comment]
please lmk when the PR is ready for review!

File: python/tvm/auto_scheduler/search_task.py [hunk 1]
```diff
@@ -481,7 +481,7 @@ def __init__(
             desc,
         )
 
-    def tune(self, tuning_options, search_policy=None):
+    def tune(self, tuning_options, search_policy=None, adaptive_training=False):
```
Comment (junrushao1994) [inline python/tvm/auto_scheduler/search_task.py:484]: probably we dont need to adjust auto_scheduler's interface

**Comment by junrushao1994** [comment]
Otherwise LGTM

File: python/tvm/auto_scheduler/search_task.py (hunk 1 above)
Comment (zxybazh) [reply python/tvm/auto_scheduler/search_task.py:484]: This only impacts `tune_te`, which does not call thru `TaskScheduler.tune` but `SearchTask.tune` instead. If we don't want to enable that for `tune_te` I can just remove that change.

Comment (junrushao1994) [reply python/tvm/auto_scheduler/search_task.py:484]: i see...that makes sense. let's keep the flag here for tune_te then :-)

**Review by junrushao1994 — APPROVED** [review]

**Pull request merged**